site/src/pages/UserSettingsPage/OAuth2ProviderPage/OAuth2ProviderPageView.tsx

```diff
@@ -0,0 +1,95 @@
+import Button from "@mui/material/Button";
+import Table from "@mui/material/Table";
+import TableBody from "@mui/material/TableBody";
+import TableCell from "@mui/material/TableCell";
+import TableContainer from "@mui/material/TableContainer";
+import TableHead from "@mui/material/TableHead";
+import TableRow from "@mui/material/TableRow";
+import { type FC } from "react";
+import type * as TypesGen from "api/typesGenerated";
+import { AvatarData } from "components/AvatarData/AvatarData";
+import { Avatar } from "components/Avatar/Avatar";
+import { ErrorAlert } from "components/Alert/ErrorAlert";
+import { TableLoader } from "components/TableLoader/TableLoader";
+
+export type OAuth2ProviderPageViewProps = {
+  isLoading: boolean;
+  error?: unknown;
+  apps?: TypesGen.OAuth2ProviderApp[];
+  revoke: (app: TypesGen.OAuth2ProviderApp) => void;
+};
+
+const OAuth2ProviderPageView: FC<OAuth2ProviderPageViewProps> = ({
+  isLoading,
+  error,
+  apps,
+  revoke,
+}) => {
+  return (
+    <TableContainer>
+      <Table>
+        <TableHead>
+          <TableRow>
+            <TableCell width="100%">Name</TableCell>
+            <TableCell width="1%" />
+          </TableRow>
+        </TableHead>
+        <TableBody>
+          {isLoading && <TableLoader />}
+          {!isLoading &&
```

Comment:
Flagging this part of the code, too, just because my other table comment might also apply here